### DISCUSSION TITLE
I can't install Fail2ban on Centos 8

### DISCUSSION THREAD

**Discussion by aaaooosss** [fail2ban/fail2ban#3827] [General]
Hi! I'm new to Github but a regular user of Fail2ban. Currently I need to install Fail2ban on a centos 8 server. However I have no success, I tried to install it with YUM and from source, equally without success.

<details><summary>SO Version</summary>
<pre>
[root@hostname]# cat /etc/centos-release
CentOS Linux release 8.5.211
</pre>
</details>

<details><summary>ReposList</summary>
<pre>
[root@hostname]# yum repolist
repo id                                                                                                        repo name
appstream                                                                                                      CentOS Linux 8 - AppStream
baseos                                                                                                         CentOS Linux 8 - BaseOS
epel-modular                                                                                                   Extra Packages for Enterprise Linux Modular 8 - x86_64
epel-next                                                                                                      Extra Packages for Enterprise Linux 8 - Next - x86_64
extras                                                                                                         CentOS Linux 8 - Extras
vasexperts                                                                                                     vasexperts.ru products repository
zabbix                                                                                                         Zabbix Official Repository - x86_64
zabbix-non-supported                                                                                           Zabbix Official Repository non-supported - x86_64
</details>
</pre>

<details><summary>Error install fail2ban with yum</summary>
<pre>
[root@nat-7-chy-chy-1 yum.repos.d]# yum install fail2ban
Last metadata expiration check: 0:31:40 ago on Mon 02 Sep 2024 02:38:30 PM -05.
No match for argument: fail2ban
Error: Unable to find a match: fail2ban
</details>
</pre>

From source:

<details><summary>python version</summary>
<pre>
[root@nat-7-chy-chy-1 yum.repos.d]# python3
Python 3.6.8 (default, Sep 10 2021, 09:13:53) 
[GCC 8.5.0 20210514 (Red Hat 8.5.0-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> exit()
</details>
</pre>

<details><summary>Debug error install</summary>
<pre>
[root@nat-7-chy-chy-1 fail2ban-master]# python3 setup.py install --without-tests
running install
running build
running build_py
package init file 'fail2ban/compat/__init__.py' not found (or not a regular file)
running build_scripts
running install_lib
running install_data
running install_egg_info
running egg_info
writing fail2ban.egg-info/PKG-INFO
writing dependency_links to fail2ban.egg-info/dependency_links.txt
writing top-level names to fail2ban.egg-info/top_level.txt
reading manifest file 'fail2ban.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'fail2ban.egg-info/SOURCES.txt'
removing '/usr/local/lib/python3.6/site-packages/fail2ban-1.1.1.dev1-py3.6.egg-info' (and everything under it)
Copying fail2ban.egg-info to /usr/local/lib/python3.6/site-packages/fail2ban-1.1.1.dev1-py3.6.egg-info
running install_scripts
WARNING: Cannot find root-base option, check the bin-path to fail2ban-scripts in "fail2ban.service" and "fail2ban-openrc.init".
Creating build/fail2ban.service (from fail2ban.service.in): @BINDIR@ -> /usr/local/bin
Creating build/fail2ban-openrc.init (from fail2ban-openrc.init.in): @BINDIR@ -> /usr/local/bin
creating fail2ban-python binding -> /usr/local/bin
changing mode of /usr/local/bin/fail2ban-client to 755
changing mode of /usr/local/bin/fail2ban-server to 755
changing mode of /usr/local/bin/fail2ban-regex to 755
changing mode of /usr/local/bin/fail2ban-testcases to 755

Please do not forget to update your configuration files.
They are in "/etc/fail2ban/".

You can also install systemd service-unit file from "build/fail2ban.service"
resp. corresponding init script from "files/*-initd".

[root@nat-7-chy-chy-1 fail2ban-master]# 
</details>
</pre>

I have been surfing the internet looking for a solution, but I have not been successful. In case it can not be installed because it is an obsolete version and does not receive maintenance.

But I would like an answer to know if I continue trying to install it.

Thank you in advance

**Comment by sebres** [comment]
> Error install fail2ban with yum

It is not part of this repository (matter of maintainer for your distro), but...
I guess you have to install epel-release firstly (see https://github.com/fail2ban/fail2ban/wiki/How-to-install-fail2ban-packages#centos--centos-stream--rhel-based-operating-systems)

> Debug error install
> From source:

I don't see any error here, sorry... It seems to be installed successful... what you'd additionally need is to create systemd-unit or init-script to start it (see https://github.com/fail2ban/fail2ban/wiki/How-to-install-or-upgrade-fail2ban-manually#manual-installation-from-source), but fail2ban-server, fail2ban-client and fail2ban-regex must be already there,
Try `fail2ban-server --version` to get its version. Or `fail2ban-client --help` for all commands.

And of course if you're not familiar, better install distribution package instead.
